src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedOption, setSelectedOption] = useState('all');
+  const [query, setQuery] = useState('');
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+
+  function handleShowTodoModalClick(todoId: number) {
+    setIsModalVisible(true);
+    setSelectedTodoId(todoId);
+  }
+
+  function handleCloseTodoModalClick() {
+    setIsModalVisible(false);
+    setSelectedTodoId(0);
+  }
+
+  useEffect(() => {
+    getTodos().then((todosFromServer) => {
+      const filteredByQuery = todosFromServer
+        .filter(todo => todo.title.toLowerCase().includes(query.toLowerCase().trim()));
+
+      switch (selectedOption) {
+        case 'active':
+          setTodos(filteredByQuery.filter(todo => !todo.completed));
+          break;
+
+        case 'completed':
+          setTodos(filteredByQuery.filter(todo => todo.completed));
+          break;
+
+        default:
+          setTodos(filteredByQuery);
+      }
```

Comment:
Put this outside of `useEffect`. because what you currently do is you send the request when the query or selectedOption changes. In this task you should load todos only when App is loaded for the first time